go/vt/vtgate/evalengine/arena.go

```diff
@@ -0,0 +1,81 @@
+package evalengine
+
+import "vitess.io/vitess/go/vt/vtgate/evalengine/internal/decimal"
+
+type Arena struct {
+	aInt64   []evalInt64
+	aUint64  []evalUint64
+	aFloat64 []evalFloat
+	aDecimal []evalDecimal
+	aBytes   []evalBytes
+}
+
+func (a *Arena) reset() {
+	a.aInt64 = a.aInt64[:0]
+	a.aUint64 = a.aUint64[:0]
+	a.aFloat64 = a.aFloat64[:0]
+	a.aDecimal = a.aDecimal[:0]
+	a.aBytes = a.aBytes[:0]
+}
+
+func (a *Arena) newEvalDecimalWithPrec(dec decimal.Decimal, prec int32) *evalDecimal {
+	if cap(a.aDecimal) > len(a.aDecimal) {
+		a.aDecimal = a.aDecimal[:len(a.aDecimal)+1]
+	} else {
+		a.aDecimal = append(a.aDecimal, evalDecimal{})
```

Comment:
Ah, nice to just fall back to whatever Go's growth strategy is to not invent our own here.